### PR TITLE
Evaluate certificate chain trust using SecTrustEvaluate

### PR DESCRIPTION
**Pull request by FredericJacobs**
The current pinning approach is quite restrictive. This PR fixes the issue of pinning with signing certificates instead of direct public key pinning.
This should work for current users without changing their code.
Uses the built-in chain evaluation function.

### CODE REVIEW

**Comment by FredericJacobs** [comment]
Thoughts @natan, @alanf?

File: SocketRocket/SRWebSocket.m [hunk 1]
```diff
@@ -1383,6 +1383,16 @@ - (void)_sendFrameWithOpcode:(SROpCode)opcode data:(id)data;
     
     [self _writeData:frame];
 }
+static BOOL evaluateServerTrust(SecTrustRef serverTrust) {
+    BOOL isValid = NO;
+    SecTrustResultType result;
+
+    OSStatus status = SecTrustEvaluate(serverTrust, &result);
+    
+    isValid = ((result == kSecTrustResultUnspecified || result == kSecTrustResultProceed) && status == 0);
```
Comment (emonti) [inline SocketRocket/SRWebSocket.m]: If you are pinning, you may actually only want to consider kSecTrustResultUnspecfied valid.

SecTrustResultProceed is defined as "The user indicated that you may trust the certificate for the purposes designated in the specified policies. ". FWIW, though, these constants are confusingly named by Apple.

See: https://developer.apple.com/library/mac/documentation/Security/Reference/certifkeytrustservices/index.html#//apple_ref/c/tdef/SecTrustResultType

File: SocketRocket/SRWebSocket.m [hunk 2]
```diff
@@ -1390,26 +1400,10 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode;
...
-                        
-                        if ([trustedCertData isEqualToData:certData]) {
-                            _pinnedCertFound = YES;
-                            break;
-                        }
-                    }
-                }
-            }
+            SecTrustRef serverTrust = (__bridge SecTrustRef)[aStream propertyForKey:(__bridge id)kCFStreamPropertySSLPeerTrust];
+            SecTrustSetAnchorCertificates(serverTrust, (__bridge CFArrayRef)sslCerts);
```
Comment (emonti) [inline SocketRocket/SRWebSocket.m]: I think you also want to ignore all other root certificates with "SecTrustSetAnchorCertificatesOnly"

See:
https://developer.apple.com/library/mac/documentation/Security/Reference/certifkeytrustservices/index.html#//apple_ref/c/func/SecTrustSetAnchorCertificatesOnly

File: SocketRocket/SRWebSocket.m (hunk 1 above)
Comment (FredericJacobs) [inline SocketRocket/SRWebSocket.m]: If you're pinning a certificate that is in the user's trust store wouldn't that return a `SecTrustResultProceed`?

File: SocketRocket/SRWebSocket.m (hunk 2 above)
Comment (dfed) [inline SocketRocket/SRWebSocket.m]: @emonti is this a merge blocker? If not, this looks good to me.

**Comment by FredericJacobs** [comment]
This PR should be closed and replaced by #274

**Pull request closed without merging**